backend/src/migrations/1543523539612_init.js

```diff
@@ -0,0 +1,98 @@
+exports.shorthands = undefined;
+
+exports.up = (pgm) => {
+    pgm.createTable('users', {
+        email: { type: 'varchar(254)', primaryKey: true},
+        first_name: {type: 'varchar(30)'},
+        last_name: {type: 'varchar(150)'},
+        password: {type: 'varchar(128)'},
+        bio: {type: 'varchar(500)'}
+    });
+
+/*****************************************************************************/
+    pgm.createTable('activities', {
+        id: 'id',
+        name: {type: 'varchar(250)', notNull: true, unique: true}
+    });
+
+    pgm.createIndex('activities', 'name');
+
+/*****************************************************************************/
+    pgm.createTable('places', {
+        id: 'id',
+        country: {type: 'varchar(250)', notNull: true},
+        city: {type: 'varchar(250)', notNull: true}
+    });
+
+    pgm.addConstraint('places', 'places_event_country_city_uniq', {
+        unique: ['country', 'city']
+    });
+
+/*****************************************************************************/
+    pgm.createTable('events', {
+        id: 'id',
+        name: {type: 'varchar(250)', notNull: true},
+        description: {type: 'text'},
+        activity: {
+            type: 'integer',
+            notNull: true,
+            references: '"activities"',
+            deferrable: true,
+            deferred: true
+        },
+        place: {
+            type: 'integer',
+            notNull: true,
+            references: '"places"',
+            deferrable: true,
+            deferred: true
+        },
+        date_from: {type: 'datetime', notNull: true},
+        date_to: {type: 'datetime', notNull: true},
+        minpeople: {
+            type: 'integer',
+            notNull: true,
+            default: 2
+        },
+        maxpeople: {
+            type: 'integer'
+        }
+    });
+
+/*****************************************************************************/
+    // Bridge table for many-to-many relationsheep between `events` and `users`
```

Comment:
What is the advantage of using a bridge table over having a users foreign key in the events table and vice versa?

Reply:
This is many-to-many relationship, it is used in RDBs in cases as ours. We have many `Users` and many `Events` arbitrary interconnected with each other. From this table we can quickly find all users attending the same event, and all events subscribed by user.

Reply:
Ok sure